pyproject.toml

```diff
@@ -35,7 +35,7 @@ pytest = "^7.1.3"
 black = "^22.8.0"
 isort = "^5.10.1"
 mypy = "^0.971"
-streamlit = "1.12.2"
+streamlit = ">=1.12.2"
```

Comment:
Let me give a quick try to see if I can fix the gallery to work with >= 1.12.2, but if not this is probably good. I would like to label the package as working with more than just 1.12.2, even though the gallery currently only works with that version.

Reply:
@arnaudmiribel Give that a try now -- now both the main code and the gallery should work with 1.14.0 -- I've pinned the gallery to 1.14.0 for now. Maybe this is just kicking the can down the road, but the other way to work with this is to actually have a separate environment locally just for the gallery (easy to do with direnv), to simulate what happens on Cloud.

Reply:
Beautiful. Reflected your changes using `poetry update` and local deployment works now. [Cloud deployment](https://arnaudmiribel-streamlit-extras-gallerystreamlit-app-upda-ze7msu.streamlitapp.com/Switch%20page%20function) also works.

Yeah it would actually be cool if Streamlit Cloud were able to take in a specific env within a pyproject.toml. 
For instance here, it could just take the `dev` one. I could see Streamlit empowering a lot of galleries or... docs.